src/renderer/components/ReactFlowBox.tsx

```diff
@@ -87,6 +103,11 @@ const updateZIndexes = (
             if (n.selected && !selectedIterators.has(n.parentNode)) {
                 relatedIterators.add(n.parentNode);
             }
+            // We need this for later
+            iteratorChildNodesMap.set(n.parentNode, [
+                ...(iteratorChildNodesMap.get(n.parentNode) ?? []),
+                n.id,
+            ]);
```

Comment:
Remember the total O(n) time complexity? For a chain with one iterator and n child nodes, this is O(n^2).
```suggestion
            let childNodes = iteratorChildNodesMap.get(n.parentNode);
            if (childNodes === undefine) {
                iteratorChildNodesMap.set(n.parentNode, childNodes = []);
            }
            childNodes.push(n.id);
```